File: parse-pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');

const dataBuffer = fs.readFileSync('public/Untitled document.pdf');

pdf(dataBuffer).then(function(data) {
    fs.writeFileSync('blogs-extracted.txt', data.text);
    console.log("Extraction complete.");
}).catch(function(error) {
    console.error("Error extracting:", error);
});

File: parse-pdf.py
import sys
try:
    import PyPDF2
except ImportError:
    import subprocess
    subprocess.check_call([sys.executable, "-m", "pip", "install", "pypdf2"])
    import PyPDF2

def extract():
    reader = PyPDF2.PdfReader('public/Untitled document.pdf')
    text = ""
    for page in reader.pages:
        text += page.extract_text() + "\n"
    
    with open('blogs-extracted.txt', 'w', encoding='utf-8') as f:
        f.write(text)
    print("Done extracting PDF")

if __name__ == "__main__":
    extract()
